test/models/nfv/vnf_tests.rb

```diff
@@ -0,0 +1,33 @@
+require "test_helper"
+require "helpers/nfv_helper"
+
+describe "Fog::NFV[:openstack] | vnf" do
+  describe "success" do
+    before do
+      set_nfv_data
+      @vnfd = @nfv.vnfds.create(:vnfd => @vnfd_data, :auth => @auth)
+      vnf_data = {:vnfd_id => @vnfd.id, :name => 'Test'}
+      @vnfs = @nfv.vnfs.create(:vnf => vnf_data, :auth => @auth)
+    end
+
+    it "#create" do
+      @vnfs.status.must_equal "ACTIVE"
+    end
+
+    it "#update" do
+      @vnfs.vnf = {:attributes => {:config => "vdus:\n  vdu1:<sample_vdu_config> \n\n"}}
+      @vnfs.update.status.must_equal "ACTIVE"
+    end
+
+    it "#destroy" do
+      sleep(10) unless Fog.mocking?
+
+      @vnfs.destroy
+      @vnfd.destroy.must_equal true
+    end
+
+    after do
```

Comment:
I would move `after` block just after `before` block so we can see what is happening before _and_ after each test without need to read through all tests.